dialectRT.go

```diff
@@ -559,11 +541,196 @@ func (d *DynamicMessage) MarshalJSON() ([]byte, error) {
 	return json.Marshal(d.Fields)
 }
 
-//UnmarshalJSON provides a custom implementation of JSON unmarshalling. Using the DynamicMessage provided, Msgspec is used to
-//determine the individual parsing of each JSON encoded payload item into the correct Go type. It is important each type is
-//correct so that the message serializes correctly and is understood by the ROS system
+//UnmarshalJSON provides a custom implementation of JSON unmarshalling. Using the DynamicMessage provided, d.t.msg is used to
```

Comment:
This one file is getting pretty long; I think it might be better if you moved DynamicMessage into its own file?